### PR TITLE
Remove parameterization from TcpTransport

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This commit is a follow up to the work completed in #27132. Essentially
it transitions two more methods (sendMessage and getLocalAddress) from
TcpTransport to TcpChannel. With this change, there is no longer a need for
TcpTransport to be aware of the specific type of channel a transport
returns. So that class is no longer parameterized by channel type.

### CODE REVIEW

**Comment by tbrooks8** [comment]
As a note, I have realized that it does not make sense for NioChannel to extend TcpChannel. NioChannel is selectable channel that can be handled by nio. TcpChannel is a channel that provides the functionality for our tcp binary protocol. We eventually will have some http channel that is an NioChannel, but not a TcpChannel. Right now this causes some issues for parameterized listener types in the nio transport. Once this is merged, I will follow this PR up with another PR that fixes this issue.

File: test/framework/src/main/java/org/elasticsearch/transport/nio/channel/NioSocketChannel.java [hunk 1]
```diff
@@ -47,6 +49,22 @@ public NioSocketChannel(String profile, SocketChannel socketChannel, SocketSelec
     }
 
     @Override
+    public void sendMessage(BytesReference reference, ActionListener<TcpChannel> listener) {
+        // TODO: temporar
```
Comment (s1monw) [inline test/framework/src/main/java/org/elasticsearch/transport/nio/channel/NioSocketChannel.java]: can we have a more verbose TODO?

File: test/framework/src/main/java/org/elasticsearch/transport/nio/channel/NioSocketChannel.java [hunk 2]
```diff
@@ -47,6 +49,22 @@ public NioSocketChannel(String profile, SocketChannel socketChannel, SocketSelec
...
+        writeContext.sendMessage(reference, new ActionListener<NioChannel>() {
```
Comment (s1monw) [inline test/framework/src/main/java/org/elasticsearch/transport/nio/channel/NioSocketChannel.java]: can't you just use the passed in listener?

**Review by s1monw — APPROVED** [review]
so much goodness! LGTM

File: test/framework/src/main/java/org/elasticsearch/transport/nio/channel/NioSocketChannel.java (hunk 1 above)
Comment (tbrooks8) [reply test/framework/src/main/java/org/elasticsearch/transport/nio/channel/NioSocketChannel.java]: Yeah. I'm not sure what happened to the rest of the comment.

File: test/framework/src/main/java/org/elasticsearch/transport/nio/channel/NioSocketChannel.java (hunk 2 above)
Comment (tbrooks8) [reply test/framework/src/main/java/org/elasticsearch/transport/nio/channel/NioSocketChannel.java]: No. ActionListener<TcpChannel> != ActionListener<NioChannel> to the compiler. I will fix this in a follow up (https://github.com/elastic/elasticsearch/pull/27407#issuecomment-344734284).

**Pull request merged**